src/nvim/lua/executor.c

```diff
@@ -96,7 +96,9 @@ static int nlua_pcall(lua_State *lstate, int nargs, int nresults)
   lua_getfield(lstate, -1, "traceback");
   lua_remove(lstate, -2);
   lua_insert(lstate, -2 - nargs);
+  RedrawingDisabled++;
```

Comment:
I don't think this should be done so generally. A switch of language from something else to lua does not mean much by itself (a plugin can call a lua function from a vimscript function or back and this doesn't indicate any change of state). Though if we do this for async RPC events (which I think is open for discussion) it should perhaps be done for scheduled lua events as well.